openlibrary/components/BarcodeScanner.vue

```diff
@@ -0,0 +1,322 @@
+    flex-direction: column;
+    // stylelint-disable declaration-block-no-duplicate-properties
+    // Fallback browsers that don't support dvh
+    height: calc(100vh - 60px);
+    height: 100dvh; // stylelint-disable-line unit-no-unknown
+    // stylelint-enable declaration-block-no-duplicate-properties
+  }
+
+  #interactive.viewport {
+    position: relative;
+    flex: 1;
+    min-height: 0;
+    video, canvas {
+      width: 100%;
+      height: 100%;
+      object-fit: cover;
+    }
+    canvas {
+      position: absolute;
+      top: 0;
+      left: 0;
+    }
+
+    video[controls] + canvas {
+      pointer-events: none;
+    }
+  }
+
+  .barcodescanner__toolbar {
+    font-family: "Lucida Grande", Veranda, Geneva, Helvetica, Arial, sans-serif;
+    position: relative;
+    background: hsl(0, 0%, 100%);
+  }
+
+  .glass-button {
+    cursor: pointer;
+
+    border-radius: 10px;
+    // stylelint-disable-next-line sh-waqar/declaration-use-variable
+    color: rgba(255,255,255, .8);
+    background: rgba(0,0,0,.5);
+    border: 1px solid rgba(255,255,255, .45);
+    font-size: .8em;
+    backdrop-filter: blur(8px);
+
+    box-shadow: 0 0 3px 0 rgba(0,0,0,.4);
+    &:disabled {
+      opacity: .5;
+    }
+  }
+
+  .icon-button {
+    display: flex;
+    align-items: center;
+    padding-right: 10px;
+  }
+
+  details.barcodescanner__advanced {
+    position: absolute;
+    bottom: 100%;
+    right: 0;
+
+    // For slide up effect
+    transform: translateY(100%);
+    transition: transform .2s;
+    > :not(summary) {
+      transition: opacity .2s;
+      opacity: 0;
+    }
+    &[open] {
+      transform: translateY(0);
+    }
+    &[open] > :not(summary) {
+      opacity: 1;
+    }
+
+    & > summary {
+      position: absolute;
+      bottom: 100%;
+      right: 0;
+      margin: 5px;
+
+      &::marker {
+        content: none;
+      }
+
+      &::-webkit-details-marker {
+        display: none;
+      }
+
+      svg {
+        width: 32px;
+        height: 32px;
+        padding: 6px;
+
+        transition: transform .2s;
+      }
+    }
+
+    &[open] > summary svg,
+    &:not([open]) > summary:hover svg {
+      transform: rotate(20deg);
+    }
+
+    // stylelint-disable-next-line no-descending-specificity
+    &[open] > summary:hover svg {
+      transform: rotate(-20deg);
+    }
+  }
+
+  .barcodescanner__controls {
+    display: flex;
+    padding: 4px;
+    padding-top: 0;
+    overflow-x: auto;
+
+    & > button {
+      padding: 8px;
+      text-align: left;
+    }
+  }
+
+  .barcodescanner__result-strip {
+    display: flex;
+    overflow-x: auto;
+    overflow-y: hidden;
+    padding: 10px;
+    .empty {
+      display: none;
+      &:first-child:last-child { display: flex; }
+      width: 80vw;
+      max-width: 300px;
+      height: 80px;
+      border: 1px dashed;
+      justify-content: center;
+      align-items: center;
+      flex-shrink: 0;
+      border-radius: 4px;
+    }
+  }
+  a.lazy-book-card {
```

Comment:
Move this into the LazyBookCard .vue file (and any other LazyBookCard-specific styling).